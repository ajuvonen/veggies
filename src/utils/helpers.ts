import {useMemoize} from '@vueuse/core';
import type {ChartOptions, ChartType, Scale} from 'chart.js';
import type {Context} from 'chartjs-plugin-datalabels';
import {DateTime} from 'luxon';
import {mergeDeep, sample} from 'remeda';
import {
  ALL_VEGGIES,
  BEANS,
  CATEGORY_EMOJI,
  COLORS,
  FRUITS,
  GRAINS,
  LEAFIES,
  MUSHROOMS,
  ROOTS,
  VEGETABLES,
} from '@/utils/constants';
import {AchievementLevel, Category} from '@/utils/types';

export const getCategoryForVeggie = useMemoize((veggie: string) => {
  if (FRUITS.includes(veggie)) {
    return Category.Fruit;
  } else if (VEGETABLES.includes(veggie)) {
    return Category.Vegetable;
  } else if (LEAFIES.includes(veggie)) {
    return Category.Leafy;
  } else if (ROOTS.includes(veggie)) {
    return Category.Root;
  } else if (BEANS.includes(veggie)) {
    return Category.Bean;
  } else if (GRAINS.includes(veggie)) {
    return Category.Grain;
  } else if (MUSHROOMS.includes(veggie)) {
    return Category.Mushroom;
  }
  return undefined;
});

export const getChartOptions = <T extends ChartType>(
  grids: boolean,
  stacked: boolean,
  withIcons: boolean,
  showChartAnimations: boolean,
  overrides: Partial<ChartOptions<T>> = {},
) =>
  mergeDeep(
    {
      animation: !showChartAnimations ? false : undefined,
      responsive: true,
      maintainAspectRatio: !grids,
      normalized: true,
      layout: {
        padding: 0,
      },
      scales: grids
        ? {
            y: {
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
            },
            y1: {
              position: 'right',
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
              afterBuildTicks: (axis: Scale) => {
                axis.ticks = [...axis.chart.scales.y.ticks];
                axis.min = axis.chart.scales.y.min;
                axis.max = axis.chart.scales.y.max;
              },
            },
            x: {
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
            },
          }
        : undefined,
      plugins: {
        title: {
          display: false,
        },
        legend: {
          display: false,
        },
        datalabels: {
          ...(withIcons
            ? {
                anchor: 'center',
                align: 'center',
                font: {
                  size: 25,
                },
                textShadowColor: '#fff',
                textShadowBlur: 5,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                formatter: (value: any, {dataset: {label}}: Context) =>
                  value ? CATEGORY_EMOJI[label as Category] : '',
              }
            : {
                display: false,
              }),
        },
        tooltip: {
          animation: showChartAnimations,
          padding: 8,
          titleFont: {
            size: 14,
            weight: 'normal',
          },
          bodyFont: {
            size: 14,
          },
          footerFont: {
            size: 14,
            weight: 'normal',
          },
          displayColors: false,
          backgroundColor: COLORS.darkBlue,
          bodyColor: COLORS.offWhite,
          titleColor: COLORS.offWhite,
        },
      },
    },
    overrides,
  ) as ChartOptions<T>;

export const getRandomVeggie = () => ALL_VEGGIES[Math.floor(Math.random() * ALL_VEGGIES.length)];

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const dateParser = (key: string, value: any) => {
  if (key === 'startDate' && value) {
    return DateTime.fromISO(value.split('T')[0]);
  }
  return value;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const dateReplacer = (key: string, value: any) => {
  if (key === 'startDate' && value) {
    return value.split('T')[0];
  }
  return value;
};

const veggieEmojis: readonly string[] = [
  '🥝',
  '🥥',
  '🍇',
  '🍈',
  '🍉',
  '🍊',
  '🍋',
  '🍌',
  '🍍',
  '🥭',
  '🍎',
  '🍏',
  '🍐',
  '🍑',
  '🍒',
  '🍓',
  '🫐',
  '🍅',
  '🍆',
  '🌽',
  '🌶️',
  '🫑',
  '🥑',
  '🥒',
  '🥬',
  '🥦',
  '🥔',
  '🧄',
  '🧅',
  '🥕',
  '🫛',
  '🥜',
  '🫘',
  '🌿',
  '🍄‍🟫',
  '🌾',
  '🌰',
];
export const getRandomEmojis = (amount: number = 1) => sample(veggieEmojis, amount);

export const achievementLevelHelper = (levels: [number, AchievementLevel][], value: number) =>
  levels.find(([threshold]) => value >= threshold)?.[1] ?? AchievementLevel.NoAchievement;
