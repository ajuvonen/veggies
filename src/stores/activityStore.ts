import {computed, ref} from 'vue';
import {defineStore, storeToRefs} from 'pinia';
import {debounceFilter, useIntervalFn, useNow, useStorage} from '@vueuse/core';
import {DateTime} from 'luxon';
import {
  countBy,
  difference,
  entries,
  intersection,
  map,
  pipe,
  prop,
  sortBy,
  take,
  takeWhile,
} from 'remeda';
import {
  Category,
  type Favorites,
  type Challenge,
  type Week,
  type Achievements,
  AchievementLevel,
} from '@/utils/types';
import {
  achievementLevelHelper,
  dateParser,
  dateReplacer,
  getCategoryForVeggie,
  getRandomVeggie,
} from '@/utils/helpers';
import {useAppStateStore} from '@/stores/appStateStore';
import {
  BEANS,
  BOTANICAL_BERRIES,
  FRUITS,
  GRAINS,
  LEAFIES,
  MUSHROOMS,
  NUTS,
  RED_VEGGIES,
  ROOTS,
  VEGETABLES,
} from '@/utils/constants';

export const useActivityStore = defineStore('activity', () => {
  const {settings} = storeToRefs(useAppStateStore());
  const currentDate = ref(DateTime.now());
  useIntervalFn(() => {
    const now = DateTime.now();
    if (!currentDate.value.hasSame(now, 'day')) {
      currentDate.value = now;
    }
  }, 2000);

  // State refs
  const startDate = useStorage<DateTime | null>('veggies-start-date', null, localStorage, {
    mergeDefaults: true,
    serializer: {
      read: (v) => (v ? DateTime.fromISO(v.split('T')[0]) : null),
      write: (v) => v?.toISODate() ?? '',
    },
  });

  const weeks = useStorage<Week[]>('veggies-weeks', [], localStorage, {
    mergeDefaults: true,
    eventFilter: debounceFilter(2000),
    serializer: {
      read: (v) => (v ? JSON.parse(v, dateParser) : null),
      write: (v) => JSON.stringify(v, dateReplacer),
    },
  });

  const challenges = useStorage<Challenge[]>('veggies-challenges', [], localStorage, {
    mergeDefaults: true,
    serializer: {
      read: (v) => (v ? JSON.parse(v, dateParser) : null),
      write: (v) => JSON.stringify(v, dateReplacer),
    },
  });

  // Computed getters
  const currentWeekStart = computed(() => currentDate.value.startOf('week'));

  const getWeekStarts = computed(() => {
    if (!startDate.value) return [currentWeekStart.value];
    const totalWeeks = Math.ceil(currentDate.value.diff(startDate.value, 'week').weeks);
    return Array.from({length: totalWeeks}, (_, weekIndex) =>
      currentWeekStart.value.minus({weeks: weekIndex}),
    );
  });

  const hotStreak = computed(() => {
    let over30Weeks = takeWhile(
      getWeekStarts.value,
      (weekStart, weekIndex) => weekIndex === 0 || veggiesForWeek.value(weekStart).length >= 30,
    ).length;
    if (currentVeggies.value.length < 30) {
      over30Weeks--;
    }
    return over30Weeks;
  });

  const allVeggies = computed(() => weeks.value.flatMap(prop('veggies')));

  const uniqueVeggies = computed(() => [...new Set(allVeggies.value)]);

  const over30Veggies = computed(
    () => weeks.value.filter(({veggies}) => veggies.length >= 30).length,
  );

  const atMostVeggies = computed(() =>
    Math.max(...weeks.value.map(({veggies}) => veggies.length), 0),
  );

  const completedChallenges = computed(
    () =>
      challenges.value.filter(({startDate, veggie}) =>
        veggiesForWeek.value(startDate).includes(veggie),
      ).length,
  );

  const veggiesForWeek = computed(
    () => (weekStart: DateTime) =>
      weeks.value.find(({startDate}) => startDate.equals(weekStart))?.veggies ?? [],
  );

  const currentVeggies = computed({
    get: () => veggiesForWeek.value(currentWeekStart.value),
    set: (veggies: string[]) => setVeggiesForWeek(veggies),
  });

  const currentChallenge = computed(
    () => challenges.value.find(({startDate}) => startDate.equals(currentWeekStart.value))?.veggie,
  );

  const suggestions = computed(() =>
    pipe(
      allVeggies.value.filter((veggie) => !currentVeggies.value.includes(veggie)),
      countBy((veggie) => veggie),
      entries(),
      sortBy([prop(1), 'desc']),
      take(settings.value.suggestionCount),
      map(prop(0)),
    ),
  );

  const favorites = computed(() =>
    Object.values(Category).reduce(
      (acc, category) => ({
        ...acc,
        [category]: pipe(
          allVeggies.value.filter((veggie) => getCategoryForVeggie(veggie) === category),
          countBy((veggie) => veggie),
          entries(),
          sortBy([prop(1), 'desc']),
          take(6),
        ),
      }),
      {} as Favorites,
    ),
  );

  const weeklyAchievements = computed(() => (veggies: string[] = currentVeggies.value) => ({
    allOnRed: achievementLevelHelper(
      [[10, AchievementLevel.Gold]],
      intersection(RED_VEGGIES, veggies).length,
    ),
    botanicalBerries: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(BOTANICAL_BERRIES, veggies).length,
    ),
    goNuts: achievementLevelHelper(
      [[5, AchievementLevel.Gold]],
      intersection(NUTS, veggies).length,
    ),
    thirtyVeggies: achievementLevelHelper(
      [
        [40, AchievementLevel.Platinum],
        [30, AchievementLevel.Gold],
      ],
      veggies.length,
    ),
  }));

  const achievements = computed<Achievements>(() => ({
    challengeAccepted: achievementLevelHelper(
      [
        [20, AchievementLevel.Gold],
        [10, AchievementLevel.Silver],
        [5, AchievementLevel.Bronze],
      ],
      completedChallenges.value,
    ),
    committed: achievementLevelHelper(
      [
        [52, AchievementLevel.Gold],
        [26, AchievementLevel.Silver],
        [12, AchievementLevel.Bronze],
      ],
      weeks.value.length,
    ),
    completionist: achievementLevelHelper(
      [
        [150, AchievementLevel.Gold],
        [80, AchievementLevel.Silver],
        [40, AchievementLevel.Bronze],
      ],
      uniqueVeggies.value.length,
    ),
    experimenterBean: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(BEANS, uniqueVeggies.value).length,
    ),
    experimenterFruit: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(FRUITS, uniqueVeggies.value).length,
    ),
    experimenterGrain: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(GRAINS, uniqueVeggies.value).length,
    ),
    experimenterLeafy: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(LEAFIES, uniqueVeggies.value).length,
    ),
    experimenterMushroom: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(MUSHROOMS, uniqueVeggies.value).length,
    ),
    experimenterRoot: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(ROOTS, uniqueVeggies.value).length,
    ),
    experimenterVegetable: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(VEGETABLES, uniqueVeggies.value).length,
    ),
    hotStreak: achievementLevelHelper(
      [
        [20, AchievementLevel.Gold],
        [10, AchievementLevel.Silver],
        [5, AchievementLevel.Bronze],
      ],
      hotStreak.value,
    ),
    thousandsEven:
      allVeggies.value.length >= 1000 && Math.floor(allVeggies.value.length / 1000) % 2 === 0
        ? AchievementLevel.Platinum
        : AchievementLevel.NoAchievement,
    thousandsOdd:
      allVeggies.value.length >= 1000 && Math.floor(allVeggies.value.length / 1000) % 2 === 1
        ? AchievementLevel.Platinum
        : AchievementLevel.NoAchievement,
    ...weeklyAchievements.value(),
  }));

  // Actions
  const toggleVeggie = (targetVeggie: string) =>
    toggleVeggieForWeek(targetVeggie, currentWeekStart.value);

  const toggleVeggieForWeek = (targetVeggie: string, weekStart: DateTime) => {
    const targetWeek = weeks.value.find(({startDate}) => startDate.equals(weekStart));
    if (!targetWeek) {
      weeks.value = [
        ...weeks.value,
        {
          startDate: weekStart,
          veggies: [targetVeggie],
        },
      ];
      challenges.value = [
        ...challenges.value,
        {
          startDate: weekStart,
          veggie: getRandomVeggie(),
        },
      ];
    } else if (!targetWeek.veggies.includes(targetVeggie)) {
      targetWeek.veggies = [...targetWeek.veggies, targetVeggie];
    } else {
      targetWeek.veggies = difference(targetWeek.veggies, [targetVeggie]);
    }
  };

  const setVeggiesForWeek = (veggies: string[], weekStart: DateTime = currentWeekStart.value) => {
    const targetWeek = weeks.value.find(({startDate}) => startDate.equals(weekStart));
    if (!targetWeek) {
      weeks.value = [
        ...weeks.value,
        {
          startDate: weekStart,
          veggies,
        },
      ];
      challenges.value = [
        ...challenges.value,
        {
          startDate: weekStart,
          veggie: getRandomVeggie(),
        },
      ];
    } else {
      targetWeek.veggies = veggies;
    }
  };

  const $reset = () => {
    startDate.value = null;
    weeks.value = [];
    challenges.value = [];
  };

  return {
    achievements,
    allVeggies,
    atMostVeggies,
    challenges,
    completedChallenges,
    currentChallenge,
    currentVeggies,
    favorites,
    getWeekStarts,
    hotStreak,
    over30Veggies,
    startDate,
    suggestions,
    uniqueVeggies,
    veggiesForWeek,
    weeklyAchievements,
    weeks,
    setVeggiesForWeek,
    toggleVeggie,
    toggleVeggieForWeek,
    $reset,
  };
});
