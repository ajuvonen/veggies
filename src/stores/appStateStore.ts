import {ref, watchEffect} from 'vue';
import {defineStore, storeToRefs} from 'pinia';
import {useStorage} from '@vueuse/core';
import type {Settings} from '@/utils/types';
import {useAchievements} from '@/hooks/achievements';
import {useActivityStore} from '@/stores/activityStore';

type Message = {
  id: string;
  text: string;
};

export const useAppStateStore = defineStore('appState', () => {
  const {advance, achievements, reset} = useAchievements();
  const {uniqueVeggies} = storeToRefs(useActivityStore());

  watchEffect(() => {
    advance(uniqueVeggies.value, 0);
  });

  // State refs
  const settings = useStorage<Settings>(
    'veggies-settings',
    {
      locale: 'en',
    },
    localStorage,
    {
      mergeDefaults: true,
    },
  );

  const messages = ref<Message[]>([]);

  const addToastMessage = (text: string) => {
    messages.value.push({
      id: crypto.randomUUID(),
      text,
    });
  };

  const removeToastMessage = (targetId: string) => {
    messages.value = messages.value.filter(({id}) => id !== targetId);
  };

  const $reset = () => {
    messages.value = [];
    settings.value = {
      locale: 'en',
    };
    reset();
  };

  return {
    messages,
    settings,
    achievements,
    addToastMessage,
    removeToastMessage,
    $reset,
  };
});
