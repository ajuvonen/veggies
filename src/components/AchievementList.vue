<script setup lang="ts">
import {storeToRefs} from 'pinia';
import {AchievementLevel, Category} from '@/utils/types';
import {useAppStateStore} from '@/stores/appStateStore';
import AchievementBadge from '@/components/AchievementBadge.vue';

const {achievements} = storeToRefs(useAppStateStore());
</script>
<template>
  <div class="achievement-list">
    <div>
      <p class="achievement-list__label">{{ $t('achievements.completionist.title') }}</p>
      <div class="achievement-list__badge-container">
        <AchievementBadge
          :active="achievements.completionist >= AchievementLevel.Bronze"
          :level="AchievementLevel.Bronze"
          achievement="completionist"
        />
        <AchievementBadge
          :active="achievements.completionist >= AchievementLevel.Silver"
          :level="AchievementLevel.Silver"
          achievement="completionist"
        />
        <AchievementBadge
          :active="achievements.completionist >= AchievementLevel.Gold"
          :level="AchievementLevel.Gold"
          achievement="completionist"
        />
      </div>
    </div>
    <div>
      <p class="achievement-list__label">{{ $t('achievements.hotStreak.title') }}</p>
      <div class="achievement-list__badge-container">
        <AchievementBadge
          :active="achievements.hotStreak >= AchievementLevel.Bronze"
          :level="AchievementLevel.Bronze"
          achievement="hotStreak"
        />
        <AchievementBadge
          :active="achievements.hotStreak >= AchievementLevel.Silver"
          :level="AchievementLevel.Silver"
          achievement="hotStreak"
        />
        <AchievementBadge
          :active="achievements.hotStreak >= AchievementLevel.Gold"
          :level="AchievementLevel.Gold"
          achievement="hotStreak"
        />
      </div>
    </div>
    <div>
      <p class="achievement-list__label">{{ $t('achievements.experimenterFruit.title') }}</p>
      <div class="achievement-list__badge-container">
        <AchievementBadge
          v-for="category in Category"
          :key="category"
          :level="AchievementLevel.Gold"
          :achievement="`experimenter${category}`"
          :active="achievements[`experimenter${category}`] === AchievementLevel.Gold"
        />
      </div>
    </div>
  </div>
</template>
<style lang="scss" scoped>
.achievement-list {
  @apply has-scroll;
  @apply flex flex-col gap-6;
}

.achievement-list__badge-container {
  @apply px-2;
  @apply flex-container justify-evenly flex-wrap;
}

.achievement-list__label {
  @apply text-xs uppercase mb-2;
}
</style>
