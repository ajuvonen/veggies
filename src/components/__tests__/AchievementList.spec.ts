import {describe, it, expect, beforeEach} from 'vitest';
import {mount} from '@vue/test-utils';
import {take} from 'remeda';
import {DateTime} from 'luxon';
import {useActivityStore} from '@/stores/activityStore';
import {BEANS, FRUITS, GRAINS, LEAFIES, ROOTS, VEGETABLES} from '@/utils/constants';
import AchievementList from '@/components/AchievementList.vue';

describe('AchievementList', () => {
  let activityStore: ReturnType<typeof useActivityStore>;

  beforeEach(() => {
    activityStore = useActivityStore();
  });

  it('renders', () => {
    const wrapper = mount(AchievementList);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAll('.badge[aria-disabled="true"]').length).toBe(12);
  });

  it('renders with badges enabled', () => {
    const thisWeek = DateTime.now().startOf('week');
    activityStore.startDate = thisWeek;
    activityStore.weeks.push({
      startDate: thisWeek,
      veggies: [
        ...take(FRUITS, 15),
        ...take(VEGETABLES, 15),
        ...take(LEAFIES, 15),
        ...take(BEANS, 15),
        ...take(ROOTS, 15),
        ...take(GRAINS, 15),
      ],
    });

    const wrapper = mount(AchievementList);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAll('.badge[aria-disabled="true"]').length).toBe(4);
  });
});
