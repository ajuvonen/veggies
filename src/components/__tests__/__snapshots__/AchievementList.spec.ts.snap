// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AchievementList > renders 1`] = `
"<div data-v-1c120264="" class="achievement-list">
  <div data-v-1c120264="">
    <p data-v-1c120264="" class="achievement-list__label">The Completionist</p>
    <div data-v-1c120264="" class="achievement-list__badge-container">
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="40 unique veggies" aria-label="40 unique veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Bronze"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🐣</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">40 veggies</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="80 unique veggies" aria-label="80 unique veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Silver"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🐥</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">80 veggies</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="150 unique veggies" aria-label="150 unique veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🦅</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">150 veggies</div>
      </div>
    </div>
  </div>
  <div data-v-1c120264="">
    <p data-v-1c120264="" class="achievement-list__label">Hot Streak</p>
    <div data-v-1c120264="" class="achievement-list__badge-container">
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="5 week streak of over 30 veggies" aria-label="5 week streak of over 30 veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Bronze"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🌶️</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">5 Weeks</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="10 week streak of over 30 veggies" aria-label="10 week streak of over 30 veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Silver"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🔥</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">10 Weeks</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="20 week streak of over 30 veggies" aria-label="20 week streak of over 30 veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🌋</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">20 Weeks</div>
      </div>
    </div>
  </div>
  <div data-v-1c120264="">
    <p data-v-1c120264="" class="achievement-list__label">The Experimenter</p>
    <div data-v-1c120264="" class="achievement-list__badge-container">
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="15 unique fruits and berries" aria-label="15 unique fruits and berries" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🍎</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Fruits</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="15 unique vegetables" aria-label="15 unique vegetables" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🥦</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Vegetables</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="15 unique leafy greens" aria-label="15 unique leafy greens" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🥬</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Leafy Greens</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="15 unique roots and bulbs" aria-label="15 unique roots and bulbs" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🥕</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Roots</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="15 unique beans and legumes" aria-label="15 unique beans and legumes" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🫛</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Beans</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="15 unique grains, nuts and seeds" aria-label="15 unique grains, nuts and seeds" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🌾</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Grains</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AchievementList > renders with badges enabled 1`] = `
"<div data-v-1c120264="" class="achievement-list">
  <div data-v-1c120264="">
    <p data-v-1c120264="" class="achievement-list__label">The Completionist</p>
    <div data-v-1c120264="" class="achievement-list__badge-container">
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="40 unique veggies" aria-label="40 unique veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Bronze"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🐣</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">40 veggies</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="80 unique veggies" aria-label="80 unique veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Silver"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🐥</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">80 veggies</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="150 unique veggies" aria-label="150 unique veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🦅</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">150 veggies</div>
      </div>
    </div>
  </div>
  <div data-v-1c120264="">
    <p data-v-1c120264="" class="achievement-list__label">Hot Streak</p>
    <div data-v-1c120264="" class="achievement-list__badge-container">
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="5 week streak of over 30 veggies" aria-label="5 week streak of over 30 veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Bronze"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🌶️</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">5 Weeks</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="10 week streak of over 30 veggies" aria-label="10 week streak of over 30 veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Silver"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🔥</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">10 Weeks</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="true" title="20 week streak of over 30 veggies" aria-label="20 week streak of over 30 veggies" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🌋</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">20 Weeks</div>
      </div>
    </div>
  </div>
  <div data-v-1c120264="">
    <p data-v-1c120264="" class="achievement-list__label">The Experimenter</p>
    <div data-v-1c120264="" class="achievement-list__badge-container">
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="15 unique fruits and berries" aria-label="15 unique fruits and berries" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🍎</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Fruits</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="15 unique vegetables" aria-label="15 unique vegetables" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🥦</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Vegetables</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="15 unique leafy greens" aria-label="15 unique leafy greens" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🥬</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Leafy Greens</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="15 unique roots and bulbs" aria-label="15 unique roots and bulbs" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🥕</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Roots</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="15 unique beans and legumes" aria-label="15 unique beans and legumes" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🫛</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Beans</div>
      </div>
      <div data-v-06381b3c="" data-v-1c120264="" aria-disabled="false" title="15 unique grains, nuts and seeds" aria-label="15 unique grains, nuts and seeds" class="badge" role="img">
        <div data-v-06381b3c="" class="badge__background badge__background--Gold"></div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__emoji">🌾</div>
        <div data-v-06381b3c="" aria-hidden="true" class="badge__text">Grains</div>
      </div>
    </div>
  </div>
</div>"
`;
